dfastmi/batch/ConfigurationInitializer.py

```diff
@@ -0,0 +1,309 @@
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, see <http://www.gnu.org/licenses/>.
+
+contact: delft3d.support@deltares.nl
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+import math
+from typing import List, Tuple
+
+from configparser import ConfigParser
+
+import numpy
+from dfastmi.batch.AConfigurationInitializerBase import AConfigurationInitializerBase
+from dfastmi.io.Reach import Reach
+
+from dfastmi.kernel.core import get_celerity, relax_factors
+from dfastmi.kernel.typehints import Vector
+
+
+class ConfigurationInitializer(AConfigurationInitializerBase):
+    """
+    Determine discharges, times, etc. for version 2 analysis
+    """
+    def init(
+        self,
+        reach: Reach,
+        config: ConfigParser
+    ) -> None:
+        """
+        Determine discharges, times, etc. for version 2 analysis
+
+        Arguments
+        ---------
+        reach : Reach
+            The reach we want to get the levels from.
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        
+        Return
+        ------
+        None
+        """
+
+        self._set_q_threshold(config, reach.qstagnant)
+        self._discharges = reach.hydro_q
+        self._apply_q = (True,) * len(self.discharges)
+        self._set_fraction_times(reach, self.q_threshold, self.discharges)
+
+        # determine the bed celerity based on the input settings
+        self._celerity = self.get_bed_celerity(reach, self.discharges)
+
+        self._rsigma = relax_factors(self.discharges, self.time_fractions_of_the_year, reach.qstagnant, self.celerity, reach.normal_width)
+        self._tstag = 0.0
+
+        self._n_fields = self._get_tide(reach, config)
+        self._needs_tide = reach.use_tide
+        if self.needs_tide:
+            self._tide_bc = reach.tide_boundary_condition
+
+    def _get_tide(self, reach: Reach, config: ConfigParser) -> int:
+        """
+        When using tide retrieve from configuration else return default 1
+
+        Arguments
+        ---------
+        reach : Reach
+            The reach we want to get the levels from.
+        config : ConfigParser
+            Configuration of the analysis to be run.
+        
+        Return
+        ------
+        n_fields : int
+            An int stating the number of fields
+        """
+        
+        if reach.use_tide:
+            try:
+                n_fields = int(config.get("General", "NFields", fallback=""))
+            except ValueError:
+                n_fields = 1 # Or should I raise ValueError exception?
```

Comment:
This piece of code looks a little bit weird, lets discuss this.

Reply:
e.g. as previously mentioned do we want to tackle config.getint in this issue?

Reply:
the try except still feels a little bit weird.

Reply:
Will be improved in DFAST-145